File: solutions/java/lasagna/1/src/main/java/Lasagna.java
public class Lasagna {

    public int expectedMinutesInOven() {
        return 40;
    }

    public int remainingMinutesInOven(int actualMinutes) {
        return expectedMinutesInOven() - actualMinutes;
    }

    public int preparationTimeInMinutes(int numberOfLayers) {
        return numberOfLayers * 2;
    }

    public int totalTimeInMinutes(int numberOfLayers, int elapsedMinutes) {
        return preparationTimeInMinutes(numberOfLayers) + elapsedMinutes;
    }
}

File: solutions/swift/lasagna/1/Sources/Lasagna/Lasagna.swift
let expectedMinutesInOven = 40
let prepTimeByLayer = 2

func remainingMinutesInOven(elapsedMinutes: Int) -> Int {
    expectedMinutesInOven - elapsedMinutes
}

func preparationTimeInMinutes(layers: Int) -> Int {
    prepTimeByLayer * layers
}

// TODO: define the 'totalTimeInMinutes' function
func totalTimeInMinutes(layers: Int, elapsedMinutes: Int) -> Int {
    preparationTimeInMinutes(layers: layers) + elapsedMinutes
}